Use root prefix to access crates

I'm working on a project which defines a `core` module for itself in the codebase and does not use the `core` crate directly. The module is frequently used, so it's imported alongside other common stuff everywhere through `use crate::*`.

However, when I tried using `serde_repr` I hit an strange error:
```
error[E0433]: failed to resolve: could not find `result` in `core`
 --> src/lib.rs:7:26
  |
7 | #[derive(Serialize_repr, Deserialize_repr, PartialEq, Eq, Debug)]
  |                          ^^^^^^^^^^^^^^^^ could not find `result` in `core`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
  |
1 | use core::fmt::Result;
  |
1 | use core::result::Result;
  |
1 | use serde::__private::Result;
  |
1 | use serde::__private::fmt::Result;
  |
    and 4 other candidates
```

After taking a look in the source code it turns out no `::` prefix is being used (e.g. `::core::result::Result` instead of `core::result::Result`), which seems to be the cause.

To reproduce:
```rust
use serde_repr::{Serialize_repr, Deserialize_repr};

mod core {}

#[derive(Serialize_repr, Deserialize_repr, PartialEq, Eq, Debug)]
#[repr(u8)]
enum E { V = 0 }
```

A temporary fix I currently use in the modules I use `serde_repr`:
```
use ::core;
```
Which is less than ideal.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -63,7 +63,7 @@ pub fn derive_serialize(input: TokenStream) -> TokenStream {
     TokenStream::from(quote! {
         impl serde::Serialize for #ident {
             #[allow(clippy::use_self)]
-            fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+            fn serialize<S>(&self, serializer: S) -> ::core::result::Result<S::Ok, S::Error>
             where
                 S: serde::Serializer
             {
@@ -93,7 +93,7 @@ pub fn derive_deserialize(input: TokenStream) -> TokenStream {
     let match_discriminants = input.variants.iter().map(|variant| {
         let variant = &variant.ident;
         quote! {
-            discriminant::#variant => core::result::Result::Ok(#ident::#variant),
+            discriminant::#variant => ::core::result::Result::Ok(#ident::#variant),
         }
     });
 
@@ -107,11 +107,11 @@ pub fn derive_deserialize(input: TokenStream) -> TokenStream {
         Some(variant) => {
             let variant = &variant.ident;
             quote! {
-                core::result::Result::Ok(#ident::#variant)
+                ::core::result::Result::Ok(#ident::#variant)
             }
         }
         None => quote! {
-            core::result::Result::Err(serde::de::Error::custom(
+            ::core::result::Result::Err(serde::de::Error::custom(
                 format_args!(#error_format, other #(, discriminant::#variants)*)
             ))
         },
@@ -120,7 +120,7 @@ pub fn derive_deserialize(input: TokenStream) -> TokenStream {
     TokenStream::from(quote! {
         impl<'de> serde::Deserialize<'de> for #ident {
             #[allow(clippy::use_self)]
-            fn deserialize<D>(deserializer: D) -> core::result::Result<Self, D::Error>
+            fn deserialize<D>(deserializer: D) -> ::core::result::Result<Self, D::Error>
             where
                 D: serde::Deserializer<'de>,
             {

